apps/api/src/server/app/middleware/error-handler.js

```diff
@@ -15,4 +15,19 @@ function errorHandler(error, _request, response, next) {
 	return response.send({ error: error.message });
 }
 
-export default errorHandler;
+/**
+ * Handle any requests thrown by failed state transitions within the state machine.
+ *
+ * @type {import('express').ErrorRequestHandler}
+ */
+export const stateMachineErrorHandler = (error, _, response, next) => {
+	if (error instanceof TransitionStateError) {
+		response.status(409).send({
+			errors: {
+				appeal: error.message
+			}
+		});
+	} else {
+		next();
+	}
```

Comment:
does there need to be an `else`? I would have thought that 
```
if (...) {
    response.send() 
}
next()
``` 
would produce the same results?

Reply:
no... response would get sent, but then the code would continue and `next` would get called and the middleware chain would continue, likely to the final error handler which then attempts to send a 500, and we'd get "Headers already sent" error or something.

Basically, we should never call next if this is our final action of the request

Reply:
so would using `return response.send()` do what i think it will?

Reply:
Return `response.send()` is another way to do this yes, but our (stupid) eslint rules then see that there's a return and want you to add returns for all if conditions, so you'd have to write:

```
if (...) {
    return response.send() 
}
return next();
```
IMO, that's uglier than an if/else, and somewhat inverts the assumed return type of {void}.